File: src/main/java/org/flighthub/Repository/ClientRepository.java
package org.flighthub.Repository;

import org.flighthub.Domain.Client;
import org.flighthub.Utils.JdbcUtils;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import java.util.Optional;
import java.util.UUID;

public class ClientRepository implements IClientRepository{
    private final JdbcUtils JDBCconnection;

    public ClientRepository(JdbcUtils connection) {
        this.JDBCconnection = connection;
    }

    @Override
    public Optional<Client> findOne(UUID id) {
        try {
            Connection connection = JDBCconnection.getConnection();
            String query = "SELECT * FROM Clients WHERE id = ?";
            PreparedStatement statement = connection.prepareStatement(query);
            statement.setString(1, id.toString());
            ResultSet resultSet = statement.executeQuery();

            if (resultSet.next()) {
                Client client = new Client(resultSet.getString("name"), resultSet.getString("address"));
                client.setId(UUID.fromString(resultSet.getString("id")));
                return Optional.of(client);
            }
        } catch (SQLException e) {
            System.out.println("Error DB "+e);
        }

        return Optional.empty();
    }

    @Override
    public Iterable<Client> findAll() {
        List<Client> clients = new ArrayList<>();

        try {
            Connection connection = JDBCconnection.getConnection();
            String query = "SELECT * FROM Clients";
            PreparedStatement statement = connection.prepareStatement(query);
            ResultSet resultSet = statement.executeQuery();

            while (resultSet.next()) {
                Client client = new Client(resultSet.getString("name"), resultSet.getString("address"));
                client.setId(UUID.fromString(resultSet.getString("id")));
                clients.add(client);
            }
        } catch (SQLException e) {
            System.out.println("Error DB "+e);
        }

        return clients;
    }

    @Override
    public Client save(Client entity) {
        try {
            Connection connection = JDBCconnection.getConnection();
            String query = "INSERT INTO Clients (id, name, address) VALUES (?, ?, ?)";
            PreparedStatement statement = connection.prepareStatement(query);
            statement.setString(1, entity.getId().toString());
            statement.setString(2, entity.getName());
            statement.setString(3, entity.getAddress());
            statement.executeUpdate();
        } catch (SQLException e) {
            System.out.println("Error DB "+e);
        }

        return entity;
    }

    @Override
    public Optional<Client> delete(UUID id) {
        Optional<Client> clientOptional = findOne(id);

        if (clientOptional.isPresent()) {
            try {
                Connection connection = JDBCconnection.getConnection();
                String query = "DELETE FROM Clients WHERE id = ?";
                PreparedStatement statement = connection.prepareStatement(query);
                statement.setString(1, id.toString());
                statement.executeUpdate();
            } catch (SQLException e) {
                System.out.println("Error DB "+e);
            }
        }

        return clientOptional;
    }

    @Override
    public Optional<Client> update(Client entity) {
        Optional<Client> clientOptional = findOne(entity.getId());

        if (clientOptional.isPresent()) {
            try {
                Connection connection = JDBCconnection.getConnection();
                String query = "UPDATE Clients SET name = ?, address = ? WHERE id = ?";
                PreparedStatement statement = connection.prepareStatement(query);
                statement.setString(1, entity.getName());
                statement.setString(2, entity.getAddress());
                statement.setString(3, entity.getId().toString());
                statement.executeUpdate();
            } catch (SQLException e) {
                System.out.println("Error DB "+e);
            }
        }

        return clientOptional;
    }
}

File: Server - C#/persistence/ClientRepository.cs
using FlightHubC_.Domain;
using FlightHubC_.Utils;
//using log4net;
using System;
using System.Collections.Generic;
using System.Data.SqlClient;
using System.Security.Cryptography;

namespace FlightHubC_.Repository
{
    public class ClientRepository : IClientRepository
    {
        private readonly SqlUtils sqlUtils;
        //private static readonly log4net.ILog log = LogManager.GetLogger(typeof(Program));

        public ClientRepository(SqlUtils sqlUtils)
        {
            this.sqlUtils = sqlUtils;
        }

        public Client FindOne(Guid id)
        {
            using (SqlConnection connection = sqlUtils.GetConnection())
            {
                //log.Info("finding one Client");

                string query = "SELECT * FROM Client WHERE id = @Id";
                SqlCommand command = new SqlCommand(query, connection);
                command.Parameters.AddWithValue("@Id", id);

                connection.Open();
                SqlDataReader reader = command.ExecuteReader();

                if (reader.Read())
                {
                    string name = reader["name"].ToString();
                    string address = reader["adress"].ToString();
                    Client c = new Client(name, address);
                    c.Id = id;
                    return c;
                }

                return null;
            }
        }

        public IEnumerable<Client> FindAll()
        {
            List<Client> clients = new List<Client>();

            using (SqlConnection connection = sqlUtils.GetConnection())
            {
                //log.Info("finding all Clients");

                string query = "SELECT * FROM Client";
                SqlCommand command = new SqlCommand(query, connection);

                connection.Open();
                SqlDataReader reader = command.ExecuteReader();

                while (reader.Read())
                {
                    Guid id = reader.GetGuid(reader.GetOrdinal("id"));
                    string name = reader["name"].ToString();
                    string address = reader["adress"].ToString();
                    Client c = new Client(name, address);
                    c.Id = id;
                    clients.Add(c);
                }
            }

            return clients;
        }

        public Client Save(Client entity)
        {
            using (SqlConnection connection = sqlUtils.GetConnection())
            {
                //log.Info("saving one Clients");

                string query = "INSERT INTO Client (id, name, adress) VALUES (@Id, @Name, @Address)";
                SqlCommand command = new SqlCommand(query, connection);
                Guid guid = Guid.NewGuid();
                entity.Id = guid;
                command.Parameters.AddWithValue("@Id", entity.Id);
                command.Parameters.AddWithValue("@Name", entity.Name);
                command.Parameters.AddWithValue("@Address", entity.Address);

                connection.Open();
                command.ExecuteNonQuery();
            }

            return entity;
        }

        public Client Delete(Guid id)
        {
            Client clientToDelete = FindOne(id);

            if (clientToDelete != null)
            {
                using (SqlConnection connection = sqlUtils.GetConnection())
                {
                    //log.Info("deleting one Client");
                    string query = "DELETE FROM Client WHERE id = @Id";
                    SqlCommand command = new SqlCommand(query, connection);
                    command.Parameters.AddWithValue("@Id", id);

                    connection.Open();
                    command.ExecuteNonQuery();
                }
            }

            return clientToDelete;
        }

        public Client Update(Client entity)
        {
            using (SqlConnection connection = sqlUtils.GetConnection())
            {
                string query = "UPDATE Client SET name = @Name, adress = @Address WHERE id = @Id";
                SqlCommand command = new SqlCommand(query, connection);
                command.Parameters.AddWithValue("@Id", entity.Id);
                command.Parameters.AddWithValue("@Name", entity.Name);
                command.Parameters.AddWithValue("@Address", entity.Address);

                connection.Open();
                command.ExecuteNonQuery();
            }

            return entity;
        }
    }
}
